contribute.md

```diff
@@ -3,104 +3,345 @@ layout: contribute
 
-Any version of Node.js that is older than those listed is not supported.
+You can view our [Github Actions configuration](https://github.com/theforeman/foreman/blob/develop/.github/workflows/js_tests.yml) to see supported NodeJS versions. Any version from that list will do. For populating the node_modules folder, npm version 4 or later is required. Alternatively, [yarn](https://yarnpkg.com) can also be used.
 
-For populating the node_modules folder, npm version 4 or later is required. Alternatively,
-[yarn](https://yarnpkg.com) can also be used.
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
 
-#### Setup development environment
-* Fork [theforeman/foreman](https://github.com/theforeman/foreman) to your GitHub account. This will create a "foreman" repo under your GitHub username.
-* Clone the fork you just created to your development system:
-{% highlight bash %}
-git clone https://github.com/<username>/foreman.git
-{% endhighlight %}
-* Reference theforeman/foreman as upstream:
-{% highlight bash %}
+```
+dnf module enable nodejs:14
+dnf install nodejs
+```
+
+#### Supported PostgreSQL versions
+
+In general, the latest version of PostgreSQL will work. Minimum version is currently 10, anything newer than that will do the job.
+
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf module enable postgresql:10
+dnf install postgresql-server
+```
+
+#### Libraries and database
+
+Some Ruby libraries (rubygems) require development tools (compilers, make), libraries and header files and Chrome driver to be present on the host. Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf groupinstall "Development Tools"
+dnf install libvirt-devel postgresql-devel openssl-devel libxml2-devel sqlite-devel libxslt-devel zlib-devel readline-devel systemd-devel libcurl-devel krb5-devel
+dnf install chromedriver
+```
+
+Foreman requires PostgreSQL, even for development. SQLite is no longer supported and will not work. Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+postgresql-setup --initdb
+```
+
+Configure security, for development purposes you can turn off authentication completely and let all users in on local sockets or loopback network interfaces. If you choose to use passwords, make sure to set them later:
+
+```
+grep trust /var/lib/pgsql/data/pg_hba.conf
+local   all             all                                     trust
+host    all             all             127.0.0.1/32            trust
+host    all             all             ::1/128                 trust
+```
+
+Start the service and create databases:
+
+```
+systemctl enable --now postgresql
+su postgres -s /bin/bash
+foreman# createuser -s foreman
+foreman# createdb --owner=foreman foreman
+foreman# createdb --owner=foreman foreman-test
+foreman# exit
```

Comment:
Or you create a superuser with the same username and use that. Then you can just create the DB without changing user:
```suggestion
sudo -u postgres createuser --superuser $USERNAME
createdb foreman
createdb foreman-test
```

Reply:
Arent you missing sudo commands for each command? I will add them.

Reply:
No, I have sudo for `createuser`. Since it's a superuser, you don't need sudo for `createdb`.